### PR TITLE
chore(Makefile): update base image

### PR DESCRIPTION
**Pull request by jchauncey**
Use the new debian-base image published to us.gcr.io/k8s-artifacts-prod instead of the old location

### CODE REVIEW

**Comment by prameshj** [comment]
Thanks. Since this is a base-image change, could you run the script:

```
./image-checks.sh <image-tag> <registry-name>
```

after building/pushing these images to a dev registry? Just to make sure they startup correctly? Changes look good to me.

**Comment by jchauncey** [comment]
cc @justaugustus

**Comment by jchauncey** [comment]
@prameshj  should that script exit cleanly? Definitely seeing some runtime errors. 

```
Status: Downloaded newer image for upstream.azurecr.io/jochaunc/oss/kubernetes/k8s-dns-node-cache:1.15.13
iptables v1.8.2 (legacy): no command specified
Try `iptables -h' or 'iptables --help' for more information.
Verifying that node-cache binary exists in node-cache image
```

**Comment by prameshj** [comment]
> @prameshj should that script exit cleanly? Definitely seeing some runtime errors.
> 
> ```
> Status: Downloaded newer image for upstream.azurecr.io/jochaunc/oss/kubernetes/k8s-dns-node-cache:1.15.13
> iptables v1.8.2 (legacy): no command specified
> Try `iptables -h' or 'iptables --help' for more information.
> Verifying that node-cache binary exists in node-cache image
> ```

Those errors are fine. It will likely throw error in sidecar image too, since not all params are passed in correctly. That output looks good for node-cache image.

**Comment by prameshj** [comment]
> > @prameshj should that script exit cleanly? Definitely seeing some runtime errors.
> > ```
> > Status: Downloaded newer image for upstream.azurecr.io/jochaunc/oss/kubernetes/k8s-dns-node-cache:1.15.13
> > iptables v1.8.2 (legacy): no command specified
> > Try `iptables -h' or 'iptables --help' for more information.
> > Verifying that node-cache binary exists in node-cache image
> > ```
> 
> Those errors are fine. It will likely throw error in sidecar image too, since not all params are passed in correctly. That output looks good for node-cache image.

You can use the output of

`./image-checks.sh 1.15.13` for reference.

**Comment by jchauncey** [comment]
ok yeah the errors look similar. I saw the other pr moving dns-node-cache to use debian-iptables. Is that something we should consider in this PR?

**Comment by prameshj** [comment]
> ok yeah the errors look similar. I saw the other pr moving dns-node-cache to use debian-iptables. Is that something we should consider in this PR?

Good point, I have asked @champtar to comment on whether we can combine the changes.

**Comment by champtar** [comment]
iptables is fixed in iptables git master, but I'm waiting for a proper 1.8.5 release (that was supposed to happen this week), to then go talk to the debian folks to have the packages updated, to then go talk to k8s folks to updates the debian-iptables base image, to then update my PR
TLDR: don't wait for me :P

File: images/dnsmasq/Makefile [hunk 1]
```diff
@@ -27,9 +27,9 @@ OUTPUT_DIR := _output/$(ARCH)
 # Ensure that the docker command line supports the manifest images
 export DOCKER_CLI_EXPERIMENTAL=enabled
 
-BASEIMAGE ?= k8s.gcr.io/debian-base-$(ARCH):v2.0.0
+BASEIMAGE ?= us.gcr.io/k8s-artifacts-prod/build-image/debian-base-$(ARCH):v2.1.0
```
Comment (prameshj) [inline images/dnsmasq/Makefile:30]: Is this because k8s.gcr.io still points to google_containers? Can this be used for building from any location?

Comment (jchauncey) [reply images/dnsmasq/Makefile:30]: yeah im not sure what caused them to change the location. The us.gcr.io domain is used now for all k8s builds so Im guessing its reliable from all locations.

Comment (prameshj) [reply images/dnsmasq/Makefile:30]: @listx Could you confirm this is ok to use? 

Once this is resolved, i think this is ok to merge.

Comment (justaugustus) [reply images/dnsmasq/Makefile:30]: @prameshj -- Yep. This is the US endpoint for community-hosted images now. See this tracking issue: https://github.com/kubernetes/kubernetes/issues/58012

The VDF (vanity domain flip for k8s.gcr.io) is being tracked here: https://github.com/kubernetes/release/issues/270

The other usable endpoints are asia.gcr.io and eu.gcr.io.

Comment (prameshj) [reply images/dnsmasq/Makefile:30]: thanks. I see https://github.com/kubernetes/kubernetes/pull/90697/files uses the same us endpoint.
